generator/EnumGen.cs

```diff
@@ -25,49 +25,103 @@
 
 namespace GtkSharp.Generation {
 
-	public class EnumGen : GenBase {
+	// <enumeration name="BookmarkFileError" c:type="GBookmarkFileError" glib:error-domain="g-bookmark-file-error-quark">
+	//   <doc xml:space="preserve">Error codes returned by bookmark file parsing.</doc>
+	//   <member name = "invalid_uri" value="0" c:identifier="G_BOOKMARK_FILE_ERROR_INVALID_URI">
+	//     <doc xml:space="preserve">URI was ill-formed</doc>
+	//   </member>
+	public class EnumGen : GenBase
+	{
+		// TODO:
+		// See marshalling without casting
+
+		class EnumMember
+		{
+			public string Name { get; private set; }
+			public string Value { get; private set; }
+			bool isBitField;
+
+			public EnumMember (XmlElement elem, bool isBitField)
+			{
+				this.isBitField = isBitField;
+
+				Name = elem.GetAttribute (Constants.Name);
+				Value = elem.GetAttribute (Constants.Value);
+			}
+
+			public bool Validate ()
+			{
+				if (IsUnsigned) {
+					Value = Value.TrimEnd ('u', 'U');
+				}
+				if (isBitField) {
+					int value = int.Parse (Value);
+					Value = string.Format ("0x{0:X}", value);
+				}
+				return true;
+			}
 
-		string enum_type = String.Empty;
-		List<string> members = new List<string> ();
+			public bool IsUnsigned {
+				get {
+					return Value.EndsWith ("U", StringComparison.OrdinalIgnoreCase);
+				}
+			}
 
-		public EnumGen (XmlElement ns, XmlElement elem) : base (ns, elem)
+			public string Generate ()
+			{
+				return Name + " = " + Value + ",";

```

Comment:
Doesn't this cause an "," on the last member of the enum where there should be none?

Reply:
This does, but for example, if another enum member is added, the gen will append the trailing comma on the line before, causing diff noise.

Reply:
Alright, I didn't knew a trailing comma without a following member was valid in C#

Reply:
Yeah, it is. And for simplicity's sake, we can generate a redundant comma rather than complicate the code.